tests/unit/systems/ops/feast/test_op.py

```diff
@@ -43,16 +44,103 @@ def test_feast_config_round_trip(tmpdir):
-        feast_op = QueryFeast(*args, **kwargs)
+        feast_op = QueryFeast(*args)
 
         created_config = feast_op.export(tmpdir + "/export_path/", input_schema, output_schema)
         created_config_dict = json.loads(created_config.parameters["queryfeast"].string_value)
 
         # now mock the QueryFeast constructor so we can inspect its arguments.
         QueryFeast.from_config(created_config_dict)
-        assert qf_init.called_with(*args, **kwargs)
+        qf_init.assert_called_with(*args)
+
+
+def test_feast_from_feature_view(tmpdir):
+
+    with patch("feast.FeatureStore.__init__", MagicMock(return_value=None)), patch(
+        "feast.feature_store.Registry.__init__", MagicMock(return_value=None)
+    ), patch(
+        "merlin.systems.dag.ops.feast.QueryFeast",
+        MagicMock(side_effect=QueryFeast),
+    ) as qf_init:
+        input_source = feast.FileSource(
+            path=tmpdir,
+            event_timestamp_column="datetime",
+            created_timestamp_column="created",
+        )
+        feature_view = feast.FeatureView(
+            name="item_features",
+            entities=["item_id"],
+            ttl=timedelta(seconds=100),
+            features=[
+                feast.Feature(name="int_feature", dtype=feast.ValueType.INT32),
+                feast.Feature(name="float_feature", dtype=feast.ValueType.FLOAT),
+                feast.Feature(name="int_list_feature", dtype=feast.ValueType.INT32_LIST),
+                feast.Feature(name="float_list_feature", dtype=feast.ValueType.FLOAT_LIST),
+            ],
+            online=True,
+            input=input_source,
+            tags={},
+        )
+        fs = feast.FeatureStore("repo_path")
+        fs.repo_path = "repo_path"
+        fs._registry = feast.feature_store.Registry(None, None)
+        fs.list_entities = MagicMock(
+            return_value=[feast.Entity(name="item_id", value_type=feast.ValueType.INT32)]
+        )
+        fs.get_feature_view = MagicMock(return_value=feature_view)
+        fs._registry.get_feature_view = MagicMock(return_value=feature_view)
+
+        expected_input_schema = Schema(
+            column_schemas=[ColumnSchema(name="item_id", dtype=np.int32)]
+        )
+
+        # The expected output is all of the feature columns plus the item_id column
+        expected_output_schema = Schema(
+            column_schemas=[
+                ColumnSchema(name="prefix_int_feature", dtype=np.int32),
+                ColumnSchema(name="prefix_float_feature", dtype=np.float32),
+                ColumnSchema(
+                    name="prefix_int_list_feature_1", dtype=np.int32, is_list=True, is_ragged=True
+                ),
+                ColumnSchema(name="prefix_int_list_feature_2", dtype=np.int32, is_list=True),
+                ColumnSchema(
+                    name="prefix_float_list_feature_1",
+                    dtype=np.float32,
+                    is_list=True,
+                    is_ragged=True,
```

Comment:
I don't quite understand why `_1` is ragged but `_2` is not ragged for these multi-hot columns.

Reply:
IIRC, because `_1` has a variable length and `_2` does not, but this really should be a single ragged column in the schemas.